scripts/deployer.ts

```diff
@@ -0,0 +1,280 @@
+import 'hardhat-deploy-ethers';
+import * as fs from 'fs';
+import { ContractReceipt, ContractTransaction } from 'ethers';
+import mainnetData from '../mainnet_deployed.json';
+import testnetData from '../testnet_deployed.json';
+import { getCurrentTimestamp } from 'hardhat/internal/hardhat-network/provider/utils/getCurrentTimestamp';
+import { ethers } from 'hardhat';
+import hre from 'hardhat';
+
+interface DeployedContract {
+  abi: Array<never>;
+  bytecode: string;
+  address: string;
+  block: number;
+  url: string;
+}
+
+interface DeployedData {
+  contracts: {
+    postageStamp: DeployedContract;
+    redistribution: DeployedContract;
+    staking: DeployedContract;
+    priceOracle: DeployedContract;
+    factory: DeployedContract;
+  };
+}
+
+// URL
+const mainnetURL = 'https://gnosisscan.io/address/';
+const testnetURL = 'https://goerli.etherscan.io/address/';
+
+const blockChainVendor = hre.network.name;
+const networkID = hre.network.config.chainId;
+
+async function main(deployedData: DeployedData = testnetData) {
+  let contractData = {
+    addresses: {
+      postageStamp: '',
+      redistribution: '',
+      staking: '',
+      priceOracle: '',
+      factory: '',
+    },
+    blocks: {
+      postageStamp: 0,
+      redistribution: 0,
+      staking: 0,
+      priceOracle: 0,
+      factory: 0,
+    },
+  };
+
+  switch (blockChainVendor) {
+    case 'testnet':
+      contractData = await deployFactory(await JSON.parse(JSON.stringify(testnetData).toString()), contractData);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
So IMHO these `ts-ignore`s are avoidable. I would suggest creating interface for `contractData` variable and then use it in its declaration. The form of the testnetData/mainnetData are then same right?